tachyon/zk/transcript/poseidon_transcript.h

```diff
@@ -0,0 +1,120 @@
+#ifndef TACHYON_ZK_TRANSCRIPT_POSEIDON_TRANSCRIPT_H_
+#define TACHYON_ZK_TRANSCRIPT_POSEIDON_TRANSCRIPT_H_
+
+#include <utility>
+#include <vector>
+
+#include "tachyon/base/buffer/buffer.h"
+#include "tachyon/base/buffer/vector_buffer.h"
+#include "tachyon/crypto/hashes/sponge/poseidon/halo2_poseidon.h"
+#include "tachyon/math/elliptic_curves/affine_point.h"
+#include "tachyon/zk/transcript/transcript.h"
+
+namespace tachyon::zk {
+
+template <typename AffinePointTy>
+class PoseidonRead : public TranscriptRead<AffinePointTy> {
+ public:
+  using ScalarField = typename AffinePointTy::ScalarField;
+  using Buffer = typename tachyon::base::Buffer;
+
+  constexpr PoseidonRead() = default;
+  explicit PoseidonRead(Buffer read_buf)
```

Comment:
I thought it was a viable change, but I'm getting such error on macos
```
In file included from tachyon/zk/transcript/poseidon_transcript_unittest.cc:7:
./tachyon/zk/transcript/poseidon_transcript.h:29:42: error: constexpr constructor's 1st parameter type 'tachyon::zk::PoseidonRead::Buffer' (aka 'tachyon::base::Buffer') is not a literal type
  constexpr explicit PoseidonRead(Buffer read_buf)
                                  ~~~~~~~^~~~~~~~
./tachyon/base/buffer/buffer.h:60:11: note: 'Buffer' is not literal because it has a non-trivial destructor
  virtual ~Buffer() = default;
          ^
./tachyon/base/buffer/buffer.h:60:11: note: destructor for 'Buffer' is not trivial because it is virtual
```
I'll just keep it this way.

Reply:
Then the question is why the others can be `constexpr`? In this case, why don't you remove all the `constexpr` from this class?

Reply:
@TomTaehoonKim Could you check out this again?